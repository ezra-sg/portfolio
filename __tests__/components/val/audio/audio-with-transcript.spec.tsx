import { renderWithLanguage } from '@/__tests__/testing-helpers';

import AudioWithTranscript from '@/components/val/audio/audio-with-transcript';

jest.mock('react-markdown', () => (
    function ReactMarkdown({ children }: { children: string }) {
        return <div>{children}</div>;
    })
);
jest.mock('@/components/val/modal/modal', () => (
    function Modal({ children }: { children: string }) {
        return <div>{children}</div>;
    })
);
jest.mock('@/components/val/audio/audio-player', () => (
    function AudioPlayer({ children }: { children: string }) {
        return <div>{children}</div>;
    })
);

describe('AudioWithTranscript', () => {
    it('should render properly', () => {
        const result = renderWithLanguage((
            <AudioWithTranscript
                description="description text"
                title="title text"
                transcript={['transcript text', 'transcript text 2']}
                src="src text"
            />
        ));

        expect(result.container).toMatchSnapshot();
    });
});
