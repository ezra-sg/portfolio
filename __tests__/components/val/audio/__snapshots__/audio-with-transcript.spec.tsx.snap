// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioWithTranscript should render properly 1`] = `
<div>
  <div
    class="flex justify-center items-center flex-col gap-2"
  >
    <h3
      aria-hidden="true"
      class="font-header dark:text-amber-50"
      id="audio-title--title-text"
    >
      title text
    </h3>
    <div />
    <div>
      <div>
        transcript text
      </div>
      <div>
        transcript text 2
      </div>
    </div>
  </div>
</div>
`;
